Broken build on macOS Sequoia

Hi @Alex2772,

I tried to build the example according to your documentation (Getting Started > Examples), but unfortunately it fails with:

```
[Building ZLIB] ninja: build stopped: subcommand failed.
CMake Error at aui.boot.cmake:1016 (message):
  Dependency build failed: ZLIB
```

Hi @Alex2772,

I'm working on a MacBook M1 with latest macOS 15.4.1. I followed the instruction to build the examples:

```
git clone https://github.com/aui-framework/aui
cd aui
mkdir build
cd build
cmake .. -DAUI_BUILD_EXAMPLES=TRUE
```

`cmake` output is:

```
-- The C compiler identification is AppleClang 17.0.0.17000013
-- The CXX compiler identification is AppleClang 17.0.0.17000013
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- AUI.Boot cache: /Users/oliver/.aui
-- AUI.Boot target ABI: appleclang-arm64
-- CMake Version: 3.31.6
-- ZLIB: resolving because /Users/oliver/.aui/prefix/zlib/55b894d3a7bb3d277c951fde7ea83b4e/INSTALLED does not exist
-- Fetching ZLIB (latest)
-- GitHub detected, checking for precompiled package...
-- Precompiled binary for ZLIB is not available
-- /Users/oliver/.aui/repo/zlib/src: checkout'ed 
-- Compiling ZLIB
Configuring CMake ZLIB:/opt/homebrew/bin/cmake /Users/oliver/.aui/repo/zlib/src -DAUI_BOOT=TRUE;--no-warn-unused-cli;-DCMAKE_INSTALL_PREFIX:PATH=/Users/oliver/.aui/prefix/zlib/55b894d3a7bb3d277c951fde7ea83b4e;-G;Unix Makefiles;-DAUIB_PRODUCED_PACKAGES_SELF_SUFFICIENT=OFF;-DAUIB_NO_PRECOMPILED=OFF;-DAUIB_FORCE_PRECOMPILED=OFF;-DAUIB_TRACE_BUILD_SYSTEM=;-DAUIB_SKIP_REPOSITORY_WAIT=;-DAUIB_CACHE_DIR=/Users/oliver/.aui;-DCMAKE_VS_PLATFORM_NAME=;-DCMAKE_BUILD_TYPE=Debug;-DCMAKE_CONFIGURATION_TYPES=;-DCMAKE_CROSSCOMPILING=FALSE;-DCMAKE_CROSSCOMPILING_EMULATOR=;-DCMAKE_MACOSX_RPATH=1;-DCMAKE_SYSTEM_NAME=Darwin;-DCMAKE_SYSTEM_VERSION=24.4.0;-DCMAKE_SYSTEM_PROCESSOR=arm64;-DCMAKE_INSTALL_NAME_DIR=@rpath;-DCMAKE_INSTALL_RPATH=@loader_path/../lib;-DCMAKE_MAKE_PROGRAM=/usr/bin/make;-DCMAKE_MSVC_RUNTIME_LIBRARY=;-DCMAKE_FIND_PACKAGE_PREFER_CONFIG=;-DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=;-DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=;-DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=;-DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=;-DONLY_CMAKE_FIND_ROOT_PATH=;-DDEPLOYMENT_TARGET=;-DCMAKE_POLICY_DEFAULT_CMP0074=NEW;-DPLATFORM=;-DXCODE_VERSION=;-DSDK_VERSION=;-DAPPLE_TARGET_TRIPLE=;-DAUI_IOS_CODE_SIGNING_REQUIRED=;-DCMAKE_POSITION_INDEPENDENT_CODE=;-DAUI_BUILD_EXAMPLES=TRUE;-DAUI_IOS_CODE_SIGNING_REQUIRED=;-DBUILD_SHARED_LIBS=ON
[Configuring ZLIB] Not searching for unused variables given on the command line.
[Configuring ZLIB] -- The C compiler identification is AppleClang 17.0.0.17000013
[Configuring ZLIB] -- The CXX compiler identification is AppleClang 17.0.0.17000013
[Configuring ZLIB] -- Detecting C compiler ABI info
[Configuring ZLIB] -- Detecting C compiler ABI info - done
[Configuring ZLIB] -- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
[Configuring ZLIB] -- Detecting C compile features
[Configuring ZLIB] -- Detecting C compile features - done
[Configuring ZLIB] -- Detecting CXX compiler ABI info
[Configuring ZLIB] -- Detecting CXX compiler ABI info - done
[Configuring ZLIB] -- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
[Configuring ZLIB] -- Detecting CXX compile features
[Configuring ZLIB] -- Detecting CXX compile features - done
[Configuring ZLIB] -- Looking for sys/types.h
[Configuring ZLIB] -- Looking for sys/types.h - found
[Configuring ZLIB] -- Looking for stdint.h
[Configuring ZLIB] -- Looking for stdint.h - found
[Configuring ZLIB] -- Looking for stddef.h
[Configuring ZLIB] -- Looking for stddef.h - found
[Configuring ZLIB] -- Check size of off64_t
[Configuring ZLIB] -- Check size of off64_t - failed
[Configuring ZLIB] -- Looking for fseeko
[Configuring ZLIB] -- Looking for fseeko - found
[Configuring ZLIB] -- Looking for unistd.h
[Configuring ZLIB] -- Looking for unistd.h - found
[Configuring ZLIB] -- Renaming
[Configuring ZLIB] --     /Users/oliver/.aui/repo/zlib/src/zconf.h
[Configuring ZLIB] -- to 'zconf.h.included' because this file is included with zlib
[Configuring ZLIB] -- but CMake generates it automatically in the build directory.
[Configuring ZLIB] -- AUI.Boot cache: /Users/oliver/.aui
[Configuring ZLIB] -- AUI.Boot target ABI: appleclang-arm64
[Configuring ZLIB] -- [AUI.BOOT] Output precompiled archive name: zlib_darwin-appleclang-arm64-shared-debug
[Configuring ZLIB] -- Configuring done (1.2s)
[Configuring ZLIB] -- Generating done (0.0s)
[Configuring ZLIB] -- Build files have been written to: /Users/oliver/.aui/repo/zlib/build/zlib/55b894d3a7bb3d277c951fde7ea83b4e
-- Building ZLIB
/Users/oliver/.aui/repo/zlib/src/uncompr.c:27:13: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C23 [-Wdeprecated-non-prototype]
   27 | int ZEXPORT uncompress2 (dest, destLen, source, sourceLen)
      |             ^
#
# Many more warnings here ...
#
17 warnings generated.
6 warnings generated.
21 warnings generated.
10 warnings generated.
23 warnings generated.
29 warnings generated.
make[1]: *** [CMakeFiles/zlib.dir/all] Error 2
make: *** [all] Error 2
[Building ZLIB] [ 18%] Building C object CMakeFiles/zlib.dir/gzlib.c.o
[Building ZLIB] [ 31%] Building C object CMakeFiles/zlib.dir/adler32.c.o
[Building ZLIB] [ 31%] Building C object CMakeFiles/zlib.dir/inftrees.c.o
[Building ZLIB] [ 31%] Building C object CMakeFiles/zlib.dir/compress.c.o
[Building ZLIB] [ 31%] Building C object CMakeFiles/zlib.dir/deflate.c.o
[Building ZLIB] [ 43%] Building C object CMakeFiles/zlib.dir/gzread.c.o
[Building ZLIB] [ 43%] Building C object CMakeFiles/zlib.dir/crc32.c.o
[Building ZLIB] [ 50%] Building C object CMakeFiles/zlib.dir/gzclose.c.o
[Building ZLIB] [ 56%] Building C object CMakeFiles/zlib.dir/gzwrite.c.o
[Building ZLIB] [ 62%] Building C object CMakeFiles/zlib.dir/infback.c.o
[Building ZLIB] [ 68%] Building C object CMakeFiles/zlib.dir/trees.c.o
[Building ZLIB] [ 75%] Building C object CMakeFiles/zlib.dir/inflate.c.o
[Building ZLIB] [ 81%] Building C object CMakeFiles/zlib.dir/zutil.c.o
[Building ZLIB] [ 87%] Building C object CMakeFiles/zlib.dir/inffast.c.o
[Building ZLIB] [ 93%] Building C object CMakeFiles/zlib.dir/uncompr.c.o
CMake Error at aui.boot.cmake:1016 (message):
  Dependency build failed: ZLIB

  note: check logs in
  /Users/oliver/.aui/prefix/zlib/55b894d3a7bb3d277c951fde7ea83b4e
Call Stack (most recent call first):
  CMakeLists.txt:57 (auib_import)


-- Configuring incomplete, errors occurred!
```

I attached the log files in `/Users/oliver/.aui/prefix/zlib/55b894d3a7bb3d277c951fde7ea83b4e`:

[configure.log](https://github.com/user-attachments/files/20021897/configure.log)

[build.log](https://github.com/user-attachments/files/20021898/build.log)

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -112,7 +112,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [windows-2019, ubuntu-latest, macos-latest]
+        os: [windows-2019, ubuntu-latest, macos-14, macos-15]
         debug_or_release: [Debug, Release]
         shared_or_static: [shared, static]
         arch: [x86_64]
@@ -131,7 +131,10 @@ jobs:
             generator: "Ninja"
             additional_cmake_flags: ''
 
-          - os: macos-latest
+          - os: macos-14
+            generator: "Ninja"
+
+          - os: macos-15
             generator: "Ninja"
 
           - os: ubuntu-latest

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -116,6 +116,7 @@ jobs:
         os: [macos-15]
         debug_or_release: [Debug, Release]
         shared_or_static: [shared, static]
+        generator: ["Ninja"]
     steps:
       - uses: actions/checkout@v4
         with:

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -108,13 +108,13 @@ jobs:
 
   build-check:
     # some additional build checks on specific platforms
-    name: ${{ matrix.os }} ${{ matrix.shared_or_static }} ${{ matrix.debug_or_release }} ${{ matrix.arch }} ${{ matrix.compiler }}
+    name: Check build ${{ matrix.os }} ${{ matrix.shared_or_static }} ${{ matrix.debug_or_release }} ${{ matrix.arch }} ${{ matrix.compiler }}
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         os: [macos-15]
-        debug_or_release: [Debug, Release]
+        debug_or_release: [Release]
         shared_or_static: [shared, static]
         generator: ["Ninja"]
     steps:
@@ -131,11 +131,11 @@ jobs:
           path: |
             ~/.aui
 
-          key: ${{ runner.os }}-check-${{ env.cache-name }}-${{ hashFiles('**/CMakeLists.txt') }}
+          key: ${{ matrix.os }}-check-${{ env.cache-name }}-${{ hashFiles('**/CMakeLists.txt') }}
           restore-keys: |
-            ${{ runner.os }}-check-${{ env.cache-name }}-
-            ${{ runner.os }}-check-
-            ${{ runner.os }}-
+            ${{ matrix.os }}-check-${{ env.cache-name }}-
+            ${{ matrix.os }}-check-
+            ${{ matrix.os }}-
 
       - name: Install Linux dependencies
         if: runner.os == 'Linux'

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -106,6 +106,63 @@ jobs:
          path: ${{github.workspace}}/build/app_project/bin/Debug/*.tar.gz
          name: aui.example.views-debug-unsigned.app.tar.gz 
 
+  build-check:
+    # some additional build checks on specific platforms
+    name: ${{ matrix.os }} ${{ matrix.shared_or_static }} ${{ matrix.debug_or_release }} ${{ matrix.arch }} ${{ matrix.compiler }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-15]
+        debug_or_release: [Debug, Release]
+        shared_or_static: [shared, static]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Cache AUI.BOOT deps
+        id: cache-aui-boot
+        uses: actions/cache@v3
+        env:
+          cache-name: aui-boot-${{ matrix.shared_or_static }}-${{ matrix.debug_or_release }}-${{ matrix.arch }}
+        with:
+          path: |
+            ~/.aui
+
+          key: ${{ runner.os }}-check-${{ env.cache-name }}-${{ hashFiles('**/CMakeLists.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-check-${{ env.cache-name }}-
+            ${{ runner.os }}-check-
+            ${{ runner.os }}-
+
+      - name: Install Linux dependencies
+        if: runner.os == 'Linux'
+        run: sudo apt-get update && sudo apt-get install pkg-config libglew-dev zlib1g-dev libssl-dev libcrypt-dev libcurl4-openssl-dev libgtk-4-dev libadwaita-1-dev libdbus-1-dev libfontconfig-dev ninja-build libpulse-dev
+
+      - name: Install macos dependencies
+        if: runner.os == 'macOS'
+        run: brew install ninja
+
+      - name: Configure CMake
+        run: cmake -G "${{ matrix.generator }}" -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{matrix.debug_or_release}} -DBUILD_SHARED_LIBS=${{ matrix.shared_or_static == 'shared' && 'ON' || 'OFF' }} ${{matrix.additional_cmake_flags}} -DAUIB_PRODUCED_PACKAGES_SELF_SUFFICIENT=ON -DAUI_BUILD_AUDIO=OFF
+
+      - name: Build project
+        run: cmake --build ${{github.workspace}}/build --config ${{matrix.debug_or_release}}
+
+      - name: Build AUI tests
+        run: cmake --build ${{github.workspace}}/build --config ${{matrix.debug_or_release}} --target Tests
+
+      - name: Run AUI tests
+        if: runner.os == 'Windows'
+        working-directory: ${{github.workspace}}/build/bin/${{ matrix.debug_or_release }}
+        run: ${{github.workspace}}/build/bin/${{ matrix.debug_or_release }}/Tests.exe
+
+      - name: Run AUI tests
+        if: runner.os == 'Linux'
+        working-directory: ${{github.workspace}}/build/bin
+        run: ${{github.workspace}}/build/bin/Tests
+
   build-desktop:
     name: ${{ matrix.os }} ${{ matrix.shared_or_static }} ${{ matrix.debug_or_release }} ${{ matrix.arch }} ${{ matrix.compiler }}
     runs-on: ${{ matrix.os }}

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -159,7 +159,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [windows-2019, ubuntu-latest, macos-14, macos-15]
+        os: [windows-2019, ubuntu-latest, macos-14]
         debug_or_release: [Debug, Release]
         shared_or_static: [shared, static]
         arch: [x86_64]

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -181,9 +181,6 @@ jobs:
           - os: macos-14
             generator: "Ninja"
 
-          - os: macos-15
-            generator: "Ninja"
-
           - os: ubuntu-latest
             generator: "Ninja"
             additional_cmake_flags: ''

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -107,7 +107,11 @@ jobs:
          name: aui.example.views-debug-unsigned.app.tar.gz 
 
   build-check:
-    # some additional build checks on specific platforms
+    # check if a build passes:
+    # - no build cache
+    # - on specific OS configurations or versions
+    # - prevent software rot
+    # - these builds are not to be published
     name: Check build ${{ matrix.os }} ${{ matrix.shared_or_static }} ${{ matrix.debug_or_release }} ${{ matrix.arch }} ${{ matrix.compiler }}
     runs-on: ${{ matrix.os }}
     strategy:
@@ -121,21 +125,6 @@ jobs:
       - uses: actions/checkout@v4
         with:
           submodules: true
-
-      - name: Cache AUI.BOOT deps
-        id: cache-aui-boot
-        uses: actions/cache@v3
-        env:
-          cache-name: aui-boot-${{ matrix.shared_or_static }}-${{ matrix.debug_or_release }}-${{ matrix.arch }}
-        with:
-          path: |
-            ~/.aui
-
-          key: ${{ matrix.os }}-check-${{ env.cache-name }}-${{ hashFiles('**/CMakeLists.txt') }}
-          restore-keys: |
-            ${{ matrix.os }}-check-${{ env.cache-name }}-
-            ${{ matrix.os }}-check-
-            ${{ matrix.os }}-
 
       - name: Install Linux dependencies
         if: runner.os == 'Linux'

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,7 +55,8 @@ function(define_aui_component AUI_COMPONENT_NAME)
 endfunction()
 
 auib_import(ZLIB https://github.com/aui-framework/zlib
-            VERSION 448a9c76a97202a21f68bd82e453dd743e944a3a)
+            VERSION 448a9c76a97202a21f68bd82e453dd743e944a3a
+            CMAKE_ARGS -DZLIB_BUILD_EXAMPLES=OFF)
 
 # define all components
 define_aui_component(core)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,7 +54,8 @@ function(define_aui_component AUI_COMPONENT_NAME)
     add_subdirectory("aui.${AUI_COMPONENT_NAME}")
 endfunction()
 
-auib_import(ZLIB https://github.com/aui-framework/zlib)
+auib_import(ZLIB https://github.com/aui-framework/zlib
+            VERSION 2b8620ab03e464499b3f46621549b29bc51c187f)
 
 # define all components
 define_aui_component(core)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,7 +55,7 @@ function(define_aui_component AUI_COMPONENT_NAME)
 endfunction()
 
 auib_import(ZLIB https://github.com/aui-framework/zlib
-            VERSION 2b8620ab03e464499b3f46621549b29bc51c187f)
+            VERSION 448a9c76a97202a21f68bd82e453dd743e944a3a)
 
 # define all components
 define_aui_component(core)

--- a/aui.curl/CMakeLists.txt
+++ b/aui.curl/CMakeLists.txt
@@ -1,8 +1,7 @@
 cmake_minimum_required(VERSION 3.10)
 
 auib_import(CURL https://github.com/aui-framework/curl/archive/refs/heads/master.zip
-            ARCHIVE
-            PRECOMPILED_URL_PREFIX https://github.com/aui-framework/curl/releases/download/refs%2Fheads%2Fmaster)
+            ARCHIVE)
 
 aui_module(aui.curl EXPORT aui WHOLEARCHIVE)
 aui_enable_tests(aui.curl)

